staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -137,15 +139,30 @@ func ObjectReaction(tracker ObjectTracker) ReactionFunc {
 			if err != nil {
 				return true, nil, err
 			}
-			// Only supports strategic merge patch
-			// TODO: Add support for other Patch types
-			mergedByte, err := strategicpatch.StrategicMergePatch(old, action.GetPatch(), obj)
-			if err != nil {
-				return true, nil, err
-			}
-
-			if err = json.Unmarshal(mergedByte, obj); err != nil {
-				return true, nil, err
+			// Only supports strategic merge patch and JSONPatch as coded.
+			switch action.GetPatchType() {
+			case types.JSONPatchType:
+				patch, err := jsonpatch.DecodePatch(action.GetPatch())
+				if err != nil {
+					return true, nil, err
+				}
+				modified, err := patch.Apply(old)
+				if err != nil {
+					return true, nil, err
+				}
+				if err = json.Unmarshal(modified, obj); err != nil {
```

Comment:
This unfortunately only works with Unstructured objects.  Normal kube objects will fail on this because Unmarshal doesn't reset maps and a few other behaviors.  I'll take fixing this.

Reply:
Fixed in https://github.com/kubernetes/kubernetes/pull/78743